tuna/rocmlir/rocmlir_worker.py

```diff
@@ -113,30 +124,39 @@ def step(self):
     self.set_job_state('running')
 
     try:
-      retcode, cmd_output = self.run_cmd()
-    except ValueError as verr:
-      self.logger.info(verr)
-      self.set_job_state('errored', result=verr)
-    else:
-      if retcode != 0:
-        quoted_output = cmd_output.replace("'", r"\'")
-        msg = f"Error code {retcode}, output {quoted_output}"
-        self.logger.info(msg)
-        self.set_job_state('errored', result=msg)
-      else:
-        try:
-          with open(self.output_filename(), 'r', encoding='utf8') as results:
-            # https://stackoverflow.com/questions/49902843/avoid-parameter-binding-when-executing-query-with-sqlalchemy
-            string = results.read().replace(':', r'\:')
-            self.set_job_state('completed', result=string)
-            self.process_result(string)
-          os.remove(self.output_filename())
-        except OSError as exc:
-          # Most often FileNotFoundError.
-          self.logger.warning(
-              'Exception occurred while saving results of job %s:  %s',
-              self.job.id, exc)
-          self.set_job_state('errored', result=repr(exc))
+      # Retry three times in the case of unhandled exceptions, logging them.
+      for attempt in Retrying(stop=stop_after_attempt(3),
```

Comment:
Where this came from:  I was observing that out of 300-odd jobs, it wasn't unusual for one to fail around here because it couldn't find the results file that came from the tuning loop.  When I retried the job, it would work.  So let's add a retry loop.  The "tenacity" library abstracts that in a nice way, hence this Retrying object, which goes three times at random intervals before raising the original exception again.

Turns out the real problem was multiple workers on one host, and our tuning loop verifies the final results by running under rocprof, and rocprof always writes to the same default file names.  So the results would sometimes step on each other and that would kill the file and cause the failure.  The better answer is to avoid the collisions in the first place;  for now, that's by using the process ID in the filename, but eventually I'll pipe stdout and avoid named files altogether.